creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.ts

```diff
@@ -288,17 +288,19 @@ const _determineNewReplicaSet = async ({
 }: DetermineNewReplicaSetParams) => {
   const currentReplicaSet = [primary, secondary1, secondary2]
   const healthyReplicaSet = new Set(
-    currentReplicaSet.filter((node) => !unhealthyReplicasSet.has(node))
+    currentReplicaSet.filter((node) => node && !unhealthyReplicasSet.has(node))
```

Comment:
the change here is to filter out any nodes that are false-y right

Reply:
exactly

Reply:
how would you feel about making this filtering logic a bit more explicit with a loop like
```js
let healthyReplicaset = new Set()
let numberOfEmptyReplicas = 0
for (let node of nodes) {
  // node is not an empty str and is healthy
  if (node && !unhealthyReplicaSet.has(node)) healthyReplicaSet.add(node)
  // node is an empty str
  else if (node.length === 0) numberOfEmptyReplicas += 1
}

Reply:
the .filter() is cleaner imo but maybe 2 separate ones with dedicated comment e.g.
```
healthyReplicaSet = currentReplicaSet
  // ensure exists
  .filter(node => Boolean)
  // node is not marked unhealthy
  .filter(node => !unhealthyReplicasSet.has(node))
```

Reply:
changed! re-requesting review